include/nuttx/mtd/mtd.h

```diff
@@ -104,6 +104,8 @@
 #  define CONFIG_MTD_REGISTRATION   1
 #endif
 
+#define SMART_FREECOUNT_BADBLOCK  0xee
```

Comment:
I don't think mtd should now about "SMART".  Low level flash driver should not know which file system is using it. How about change it to "MTD_BADBLOCK_MARK"?

Reply:
OK